addOns/pscanrulesAlpha/src/test/java/org/zaproxy/zap/extension/pscanrulesAlpha/InsecureComponentScannerUnitTest.java

```diff
@@ -0,0 +1,408 @@
+
+        // When
+        rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+        // Then
+        assertVulnerabilityAlertRaised(Alert.RISK_LOW, "Servlet A JBoss-5.0.0");
+    }
+
+    @Test
+    public void shouldNotRaiseVulnerabilityAlertWhenSecureJBossWebServerUsed()
+            throws URIException, HttpMalformedHeaderException {
+        // Given
+        HttpMessage msg = createMessage();
+        msg.setResponseHeader("HTTP/1.1 200 OK\r\n" + "Server: Servlet A JBoss-5.0.1");
+
+        // When
+        rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+        // Then
+        assertThat(alertsRaised.size(), equalTo(0));
+    }
+
+    @Test
+    public void shouldRaiseVulnerabilityAlertWhenInsecureApacheTomcatWebServerUsed()
+            throws URIException, HttpMalformedHeaderException {
+        // Given
+        HttpMessage msg = createMessage();
+        msg.setResponseHeader("HTTP/1.1 200 OK\r\n" + "Server: example");
+        msg.setResponseBody("<html><head><title>Apache Tomcat/9.1.0 -</title>");
+
+        // When
+        rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+        // Then
+        assertVulnerabilityAlertRaised(Alert.RISK_LOW, msg.getResponseBody().toString());
+    }
+
+    @Test
+    public void shouldNotRaiseVulnerabilityAlertWhenSecureApacheTomcatWebServerUsed()
+            throws URIException, HttpMalformedHeaderException {
+        // Given
+        HttpMessage msg = createMessage();
+        msg.setResponseHeader("HTTP/1.1 200 OK\r\n" + "Server: example");
+        msg.setResponseBody("<html><head><title>Apache Tomcat/9.1.2 -</title>");
+
+        // When
+        rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+        // Then
+        assertThat(alertsRaised.size(), equalTo(0));
+    }
+
+    @Test
+    public void shouldRaiseVulnerabilityAlertWhenInsecureProxyServerUsed()
+            throws URIException, HttpMalformedHeaderException {
+        // Given
+        HttpMessage msg = createMessage();
+        msg.setResponseHeader("HTTP/1.1 200 OK\r\n" + "Via: (proxy/3.0)");
+
+        // When
+        rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+        // Then
+        assertVulnerabilityAlertRaised(Alert.RISK_LOW, "(proxy/3.0)");
+    }
+
+    @Test
+    public void shouldNotRaiseVulnerabilityAlertWhenSecureProxyServerUsed()
+            throws URIException, HttpMalformedHeaderException {
+        // Given
+        HttpMessage msg = createMessage();
+        msg.setResponseHeader("HTTP/1.1 200 OK\r\n" + "Via: (proxy/3.3)");
+
+        // When
+        rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+        // Then
+        assertThat(alertsRaised.size(), equalTo(0));
+    }
+
+    /* In the other methods the Web-Server was leaked in the "Server" headerdirective.
+      In this method, we verify, that the "X-Powered-By" headerdirective is threated equivalent
```

Comment:
Treated has a typo.

Should headerdirective be two words?

Reply:
Instead of "headerdirective" i write "header-directive"